### DISCUSSION TITLE
Can you add M3 plugins via a simple javascript file?

### DISCUSSION THREAD

**Discussion by jpadfield** [ProjectMirador/mirador#3484] [Q&A] [answered]
Hi,

It is possible to add an instance of mirador 3 to a basic webpage simply by including one javascript file like: https://cdn.jsdelivr.net/npm/mirador@3.2.0/dist/mirador.min.js

Now my question is, is it possible to add in plugins the same way? The instructions all seem to relate to downloading and building code into a project. 

For example I can see that https://www.jsdelivr.com/package/npm/mirador-image-tools is also listed on jsdeliver, but the default script does seem to deliver that simple add in and play type option.

The use case is being able to add it into webpages similar to: https://jpadfield.github.io/simple-site/Mirador%20IIIF%20viewer.html

Thanks

**Comment by jbaiter** [comment, marked as answer]
I don't think this is currently possible, I always ran into inscrutable errors regarding react/redux when I tried this with my own plugin :-/

Your best bet is currently to use a simple bundler like Parcel (should be zero config for this use case) and bundle up your custom mirador package that includes the plugins of your choice.

**Comment by jpadfield** [comment]
Ok, thanks I will have a look.

**Comment by francescopioscognamiglio** [comment]
Hi, any update to add M3 plugins via a simple javascript file?

**Comment by jpadfield** [comment]
Beyond the suggested answer I have not managed to find an alternative option. I had a quick look at Parcel, but have not used it in anger yet as it would not really work for the use case.

**Reply by francescopioscognamiglio** [reply]
Thank you for the feedback.
I am testing to use mirador plugins as simple js to be imported inside my Java web application
Then I tried to use webpack/parcel tools to generate js that I can import directly on my application, but this still not working.
